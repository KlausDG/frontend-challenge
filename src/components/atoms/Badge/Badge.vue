<script setup lang="ts">
import { Typography } from '@/components/atoms'
</script>

<template>
  <div class="badge__container">
    <Typography color="primary"><slot /></Typography>
  </div>
</template>

<style lang="scss" scoped>
.badge__container {
  border-radius: $radius-md;
  padding: $spacing-xs;
  background-color: $color-faded;
}
</style>
