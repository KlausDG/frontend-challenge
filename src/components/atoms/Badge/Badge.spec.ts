import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import Badge from './Badge.vue'

describe('Badge.vue', () => {
  it('renders the slot content correctly', () => {
    const wrapper = mount(Badge, {
      slots: {
        default: 'Badge Text',
      },
    })

    expect(wrapper.text()).toBe('Badge Text')
  })

  it('renders the Typography component inside the badge', () => {
    const wrapper = mount(Badge, {
      slots: {
        default: 'Badge Content',
      },
    })

    const typography = wrapper.findComponent({ name: 'Typography' })
    expect(typography.exists()).toBe(true)
  })

  it('has the correct CSS class on the container', () => {
    const wrapper = mount(Badge)
    expect(wrapper.classes()).toContain('badge__container')
  })

  it('applies the correct color prop to Typography', () => {
    const wrapper = mount(Badge, {
      slots: {
        default: 'Badge Text',
      },
    })

    const typography = wrapper.findComponent({ name: 'Typography' })
    expect(typography.props('color')).toBe('primary')
  })

  it('matches the component snapshot', () => {
    const wrapper = mount(Badge, {
      slots: {
        default: 'Snapshot Test',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
