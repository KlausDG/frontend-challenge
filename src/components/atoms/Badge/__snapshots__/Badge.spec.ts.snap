// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge.vue > matches the component snapshot 1`] = `
"<div data-v-df2d503b="" class="badge__container">
  <p data-v-df2d503b="" class="typography _caption_d0cbd2 _primary_d0cbd2">Snapshot Test</p>
</div>"
`;
